### PR TITLE
Update cori after upgrade/remove edison

### PR DESCRIPTION
**Pull request by milenaveneziani**
This PR does two things:
- cherry-picks #055e950 for cori-knl/cori-haswell 2019 upgrade (from recent E3SM merge into maint-1.0, E3SM PR#3097);
- removes edison from config_machines and config_compilers

### CODE REVIEW

File: cime/config/e3sm/machines/config_machines.xml [hunk 1]
```diff
@@ -55,290 +55,42 @@
...
-    <environment_variables compiler="intel">
-      <env name="FORT_BUFFERED">yes</env>
-    </environment_variables>
-    <environment_variables compiler="intel18">
-      <env name="FORT_BUFFERED">yes</env>
-    </environment_variables>
-
-  </machine>
-
-  <machine MACH="onyx">
```
Comment (milenaveneziani) [inline cime/config/e3sm/machines/config_machines.xml]: @apcraig: don't panic, onyx is still here, just further down :)

File: cime/config/e3sm/machines/config_machines.xml [hunk 2]
```diff
@@ -593,49 +344,45 @@
...
     </environment_variables>
-    <environment_variables compiler="intel19">
-      <env name="MPICH_MEMORY_REPORT">1</env>
-    </environment_variables>
   </machine>
 
   <!-- Skylake nodes of Stampede2 at TACC -->
   <machine MACH="stampede2">
-    <DESC>Stampede2. Intel skylake nodes at TACC. 48 cores per node, batch system is SLURM</DESC>
+    <DESC>Stampede2. Intel skylake nodes at TACC.  48 cores per node, batch system is SLURM</DESC>
```
Comment (milenaveneziani) [inline cime/config/e3sm/machines/config_machines.xml]: sorry, these changes and a few more below for machines that have nothing to do with NERSC, came about while resolving the conflicts of the cherry-pick.

**Review by apcraig — COMMENTED** [review]
The onyx machine changes are showing up in this PR for some reason.  That sometimes happens if the code mods are added on top of an earlier version of the branch.  Is that the case here?

**Comment by milenaveneziani** [comment]
yes, indeed. If you notice the onyx block got moved from up top to further down in the file. I can move it back up if you prefer, but it should just be a result of moving things around while trying to solve the conflicts. I think everything should be there for onyx though.

**Review by apcraig — APPROVED** [review]
Looks good, thanks.  I thought that onyx was being re-added.  I didn't notice it was removed from another part and moved.  No problem moving it.  I haven't tested, but can once this PR is committed.

**Comment by milenaveneziani** [comment]
@apcraig: Gennaro is having problems compiling with this. I told him to use his old version and just make the changes to the cori-haswell block in config_machines from Noel's commit, to see if that works.

**Comment by milenaveneziani** [comment]
ok, I re-did this in a much cleaner way. Gennaro is able to compile now.
I'll wait until he completes a 4-year run successfully before merging. It'll probably be the end of the week.

**Comment by milenaveneziani** [comment]
This seems to be working (although running on cori is far from going smoothly at this point, with node failures and such).

**Pull request merged**